sport/app/football/controllers/ResultsController.scala

```diff
@@ -2,70 +2,81 @@ package football.controllers
 
 import common.Edition
 import feed.Competitions
-import play.api.mvc.{AnyContent, Action}
+import play.api.mvc.{AnyContent, Action, Result}
 import org.joda.time.LocalDate
 import model._
 import football.model._
 import pa.FootballTeam
 import model.Competition
 
-
 object ResultsController extends MatchListController with CompetitionResultFilters {
 
-  private def results(date: LocalDate) = new ResultsList(date, Competitions())
-  private val page = new FootballPage("football/results", "football", "All results", "GFE:Football:automatic:results")
-
-  def allResultsForJson(year: String, month: String, day: String) = allResultsFor(year, month, day)
-  def allResultsFor(year: String, month: String, day: String): Action[AnyContent] =
-    renderAllResults(createDate(year, month, day))
+  private def competitionOrTeam(tag: String): Option[Either[Competition, FootballTeam]] = {
+    lookupCompetition(tag).map(Left(_))
+      .orElse(lookupTeam(tag).map(Right(_)))
+  }
 
-  def allResultsJson() = allResults()
-  def allResults(): Action[AnyContent] =
-    renderAllResults(LocalDate.now(Edition.defaultEdition.timezone))
+  private def byType[A](ifAll: => A)(ifCompetition: Competition => A)(ifTeam: FootballTeam => A)(tag: Option[String]): Option[A] = {
```

Comment:
I don't really like this method name but I couldn't find something better. Suggestion welcome :)